src/org/zaproxy/zap/extension/pscanrulesAlpha/StrictTransportSecurityScanner.java

```diff
@@ -47,9 +50,11 @@
 	//and is used to reset browser's settings for HSTS.
 	//If found raise an INFO alert.
 	//Pattern accounts for potential spaces and quotes
-	private static final Pattern BAD_MAX_AGE_PATT = Pattern.compile("max-age\\s*=\\s*\'*\"*\\s*0\\s*\"*\'*\\s*", Pattern.CASE_INSENSITIVE);
+	private static final Pattern BAD_MAX_AGE_PATT = Pattern.compile("\\bmax-age\\s*=\\s*\'*\"*\\s*0\\s*\"*\'*\\s*", Pattern.CASE_INSENSITIVE);
+	//Ensure max-age actually contains a digit
+	private static final Pattern MAX_AGE_PATT = Pattern.compile("\\bmax-age\\s*=\\s*\'*\"*\\s*\\s*\\d+\\s*\"*\'*\\s*", Pattern.CASE_INSENSITIVE);

```

Comment:
Using the RFC leads to a regex too big, better not going that way. How about just checking that is has ASCII chars (i.e. it must match `\\p{ASCII}*`)? That would catch the mentioned case (it would not catch other malformed cases, though).